client/src/contexts/SocketIOContext.tsx

```diff
@@ -2,16 +2,23 @@ import React, { createContext, useState, useEffect } from "react";
 import io, { Socket } from "socket.io-client";
 import PropTypes from "prop-types";
 
-const stripTrailingSlash = (str: string) => str.replace(/\/$/, '')
+// Get the room name from a namespace path.
+const getRoomName = (nsp: string) => {
+  // To do: replace split with path.basename if possible.
+  return encodeURIComponent(nsp.split("/")[1]);
```

Comment:
`nsp.split("/")[1]` does not seem intuitive, would be helpful to have an example how the `nsp` string is passed.

```javascript
// nsp is something like /roomname
```